solr/core/src/java/org/apache/solr/handler/admin/api/ReloadCoreAPI.java

```diff
@@ -17,52 +17,56 @@
-import java.util.Locale;
-import java.util.Map;
-import org.apache.solr.api.Command;
-import org.apache.solr.api.EndPoint;
-import org.apache.solr.api.PayloadObj;
-import org.apache.solr.common.params.CoreAdminParams;
-import org.apache.solr.common.util.ReflectMapWriter;
-import org.apache.solr.handler.admin.CoreAdminHandler;
+import io.swagger.v3.oas.annotations.Parameter;
+import javax.inject.Inject;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import org.apache.solr.api.JerseyResource;
+import org.apache.solr.core.CoreContainer;
+import org.apache.solr.jersey.PermissionName;
+import org.apache.solr.jersey.SolrJerseyResponse;
+import org.apache.solr.request.SolrQueryRequest;
+import org.apache.solr.response.SolrQueryResponse;
 
 /**
- * V2 API for reloading an individual core.
+ * V2 API for reloading collections.
  *
- * <p>The new API (POST /v2/cores/coreName {'reload': {...}}) is equivalent to the v1
- * /admin/cores?action=reload command.
- *
- * @see ReloadCorePayload
+ * <p>The new API (POST /v2/core/coreName/reload {...}) is analogous to the v1
+ * /admin/core?action=RELOAD command.
  */
-@EndPoint(
-    path = {"/cores/{core}"},
-    method = POST,
-    permission = CORE_EDIT_PERM)
-public class ReloadCoreAPI {
-  private static final String V2_RELOAD_CORE_CMD = "reload";
+@Path("/cores/{coreName}/reload")
+public class ReloadCoreAPI extends JerseyResource {
 
-  private final CoreAdminHandler coreHandler;
+  private final SolrQueryRequest solrQueryRequest;
+  private final SolrQueryResponse solrQueryResponse;
+  private final CoreContainer coreContainer;
 
-  public ReloadCoreAPI(CoreAdminHandler coreHandler) {
-    this.coreHandler = coreHandler;
+  @Inject
+  public ReloadCoreAPI(
+      SolrQueryRequest solrQueryRequest,
+      SolrQueryResponse solrQueryResponse,
+      CoreContainer coreContainer) {
+    this.solrQueryRequest = solrQueryRequest;
+    this.solrQueryResponse = solrQueryResponse;
+    this.coreContainer = coreContainer;
   }
 
-  @Command(name = V2_RELOAD_CORE_CMD)
-  public void reloadCore(PayloadObj<ReloadCorePayload> obj) throws Exception {
-    final String coreName = obj.getRequest().getPathTemplateValues().get(CoreAdminParams.CORE);
-
-    final Map<String, Object> v1Params = new HashMap<>();
-    v1Params.put(ACTION, CoreAdminParams.CoreAdminAction.RELOAD.name().toLowerCase(Locale.ROOT));
-    v1Params.put(CoreAdminParams.CORE, coreName);
-
-    coreHandler.handleRequestBody(wrapParams(obj.getRequest(), v1Params), obj.getResponse());
+  @POST
+  @Produces({"application/json", "application/xml", BINARY_CONTENT_TYPE_V2})
+  @PermissionName(CORE_EDIT_PERM)
+  public SolrJerseyResponse reloadCore(
+      @Parameter(description = "The name of the core to snapshot.", required = true)
+          @PathParam("coreName")
+          String coreName) {
+    SolrJerseyResponse solrJerseyResponse = instantiateJerseyResponse(ReloadCoreResponse.class);
```

Comment:
[0] See my comment on L71 below about whether ReloadCoreResponse is really needed, or whether we can maybe use  `SolrJerseyResponse.class` as the method arg here.